packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -458,15 +462,81 @@ class DebuggerController extends DisposableController
     }
   }
 
-  void selectStackFrame(StackFrameAndSourcePosition frame) {
+  void selectStackFrame(StackFrameAndSourcePosition frame) async {
     _selectedStackFrame.value = frame;
 
+    if (frame?.frame != null) {
+      _variables.value = await _variablesForFrame(frame.frame);
+    }
+
     if (frame?.scriptRef != null) {
       showScriptLocation(
           ScriptLocation(frame.scriptRef, location: frame.sourcePosition));
     }
   }
 
+  Future<List<Variable>> _variablesForFrame(Frame frame) async {
```

Comment:
This method is longer, w/ a few levels of nesting. Can this be broken out into a few smaller helper methods? Or, if we expect more complexity here later, perhaps into a helper class of some kind?